htdocs/mrp/index.php

```diff
@@ -147,6 +148,71 @@
 
 print '<br>';
 
+$sql = "SELECT bom.rowid, bom.ref, bom.label ";
+$sql .= " FROM ".MAIN_DB_PREFIX."bom_bom AS bom, ".MAIN_DB_PREFIX."bom_bomline AS blin, ";
+$sql .= MAIN_DB_PREFIX."entrepot AS entpt, ".MAIN_DB_PREFIX."product_stock AS stk";
+$sql .= " WHERE bom.rowid = blin.fk_bom";
+$sql .= " AND blin.fk_product = stk.fk_product";
+$sql .= " AND stk.fk_entrepot = entpt.rowid";
+$sql .= " AND entpt.statut = 1";
+$sql .= " AND bom.entity = ".getEntity('bom');
+$sql .= " AND bom.status = 1";
+$sql .= " AND bom.rowid NOT IN (";
+$sql .= " SELECT b.rowid";
+$sql .= " FROM ".MAIN_DB_PREFIX."bom_bom AS b";
+$sql .= " WHERE b.entity = ".getEntity('bom');
+$sql .= " AND b.rowid IN (";
```

Comment:
You can avoid the 
b.rowid NOT IN (SELECT ... b.rowid IN (SELECT ...
with
b.rowid NOT IN (SELECT bom_bomline AS bl, bom as b ...  and a joint between bl and b.